api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java

```diff
@@ -61,10 +62,14 @@
   @Mapping(target = "researchPurpose", source = "dbWorkspace")
   Workspace toApiWorkspace(DbWorkspace dbWorkspace);
 
-  @Mapping(target = "workspace", source = "dbWorkspace")
-  @Mapping(target = "accessLevel", source = "firecloudWorkspaceResponse")
-  WorkspaceResponse toApiWorkspaceResponse(
-      DbWorkspace dbWorkspace, FirecloudWorkspaceResponse firecloudWorkspaceResponse);
+  default WorkspaceResponse toApiWorkspaceResponse(
```

Comment:
I switched to a default mapper because apparently `@Mapping` cannot support multivariate mappers: https://github.com/mapstruct/mapstruct/issues/621

Reply:
It might be cleaner to declare `WorkspaceResponse toApiWorkspaceResponse(Workspace w, AccessLevel a)` and then build/extract the access level (or cajole the mapper to do it) when calling that. It's a more natural signature to expose, and I think we can more simply implement `WorkspaceResponse toApiWorkspaceResponse(DbWorkspace dbWorkspace, FirecloudWorkspaceResponse firecloudWorkspaceResponse)`.

Reply:
I think this is what you were suggesting. I think I agree this is slightly cleaner - at least it allows me to remove the mapper fetch.